### ISSUE TITLE
.NET 7 broke BrotliCompressor due to invalid compression levels

### ISSUE THREAD

**Issue by CumpsD** [Taritsyn/WebMarkupMin#150]
Have a look at this runtime change: https://github.com/dotnet/runtime/commit/1d08e154b942a41e72cbe044e01fff8b13c74496 from PR https://github.com/dotnet/runtime/pull/57561

They now throw on an invalid level:

```csharp
            compressionLevel switch
            {
                CompressionLevel.Optimal => Quality_Default,
                CompressionLevel.NoCompression => Quality_Min,
                CompressionLevel.Fastest => 1,
                CompressionLevel.SmallestSize => Quality_Max,
                _ => throw new ArgumentException(SR.ArgumentOutOfRange_Enum, nameof(compressionLevel))
            };
```

In the passed they simply casted it (`_ => (int)level,`)

However in `BrotliCompressor` you are doing:
https://github.com/Taritsyn/WebMarkupMin/blob/master/src/WebMarkupMin.AspNet.Brotli/BrotliCompressor.cs#L72-L91

```csharp
#if NETSTANDARD2_1
			CompressionLevel compressionLevel;

			switch (_settings.Level)
			{
				case 0:
					compressionLevel = CompressionLevel.NoCompression;
					break;
				case 1:
				case 2:
					compressionLevel = CompressionLevel.Fastest;
					break;
				case int level when level >= 3 && level <= 11:
					compressionLevel = (CompressionLevel)level;
					break;
				default:
					throw new NotSupportedException();
			}

			var brotliStream = new BrotliStream(stream, compressionLevel);
#else
			var brotliStream = new BrotliStream(stream, CompressionMode.Compress);
			brotliStream.SetQuality(_settings.Level);
#endif
```

Which means when somebody passes in a level (or uses the default of 5 from https://github.com/Taritsyn/WebMarkupMin/blob/master/src/WebMarkupMin.AspNet.Brotli/BrotliCompressionSettings.cs#L54) it crashes with the following stacktrace:

```text
System.ArgumentException: Enum value was out of legal range. (Parameter 'compressionLevel')
   at System.IO.Compression.BrotliUtils.GetQualityFromCompressionLevel(CompressionLevel compressionLevel)
   at System.IO.Compression.BrotliStream..ctor(Stream stream, CompressionLevel compressionLevel)
   at WebMarkupMin.AspNet.Brotli.BrotliCompressor.Compress(Stream stream)
   at WebMarkupMin.AspNetCore3.BodyWrapperStreamBase.InternalFinishAsync()
   at WebMarkupMin.AspNetCore3.BodyWrapperStreamWithResponseBodyFeature.FinishAsync()
   at WebMarkupMin.AspNetCore3.WebMarkupMinMiddleware.InvokeCore(HttpContext context, Boolean useMinification, Boolean useCompression)
   at WebMarkupMin.AspNetCore3.WebMarkupMinMiddleware.InvokeCore(HttpContext context, Boolean useMinification, Boolean useCompression)
   at WebMarkupMin.AspNetCore3.WebMarkupMinMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.ResponseCompression.ResponseCompressionMiddleware.InvokeCore(HttpContext context)
   at Microsoft.WebTools.BrowserLink.Net.BrowserLinkMiddleware.ExecuteWithFilterAsync(IHttpSocketAdapter injectScriptSocket, String requestId, HttpContext httpContext)
   at Microsoft.AspNetCore.Watch.BrowserRefresh.BrowserRefreshMiddleware.InvokeAsync(HttpContext context)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application)
```

Right now there is no way to end at `CompressionLevel.Optimal` (which is value `0` in the enum) since that gets parsed to `CompressionLevel.NoCompression`

https://learn.microsoft.com/en-us/dotnet/api/system.io.compression.compressionlevel?view=net-7.0

It seems it would be easiest to expose the .NET surface in WebMarkupMin instead?

**Comment by Taritsyn** [comment]
Hello, David!

Thank you for information! I will investigate this issue in the next few days.

For now, I recommend that you replace the `BrotliCompressor` from `WebMarkupMin.Brotli` namespace by the `BuiltInBrotliCompressor` from `WebMarkupMin.AspNet.Common.Compressors` namespace. Installation of additional packages is not required, and also, if you wish, can uninstall the WebMarkupMin.Brotli package.

I also recommend replacing the WebMarkupMin.AspNetCore3 package by the [WebMarkupMin.AspNetCore7](https://www.nuget.org/packages/WebMarkupMin.AspNetCore7/) package.

**Comment by CumpsD** [comment]
Thanks! I will have a look

**Comment by Taritsyn** [comment]
This error has been fixed in [version 2.13.2](https://github.com/Taritsyn/WebMarkupMin/releases/tag/v2.13.2).

**Comment by CumpsD** [comment]
Thanks! I switched to AspNetCore7 package and BuiltIn now :)
